### PR TITLE
[WIP][IMP] agreement: Minor improvements

### PR DESCRIPTION
**Pull request by sergio-teruel**

### CODE REVIEW

**Review by ygol — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@sergio-teruel is this still WIP?

**Comment by sergio-teruel** [comment]
Yes, I think... but maybe this PR can be merged...

File: agreement/security/ir.model.access.csv [hunk 1]
```diff
@@ -1,3 +1,5 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 access_agreement_read,Read access on agreement to Employees,model_agreement,base.group_user,1,0,0,0
 access_agreement_full,Full access on agreement grp,model_agreement,base.group_no_one,1,1,1,1
+access_agreement_type_read,Read access on agreement type to Employees,model_agreement_type,base.group_user,1,0,0,0
+access_agreement_type_full,Full access on agreement type grp,model_agreement_type,base.group_no_one,1,1,1,1
```
Comment (pedrobaeza) [inline agreement/security/ir.model.access.csv]: Why giving full access to developer mode? I think it should be to settings group.

**Comment by sergio-teruel** [comment]
I have kept the original line of the module

**Comment by pedrobaeza** [comment]
I think both are incorrect: it should be settings group. If not, anyone can change that switching to developer mode. @ygol @max3903 do you agree?

**Comment by pedrobaeza** [comment]
Please do the switch of the permissions as @ygol seems to agree and they are logical.

File: agreement/security/ir.model.access.csv (hunk 1 above)
Comment (sergio-teruel) [reply agreement/security/ir.model.access.csv]: Ok. What groups can I use here??

**Comment by sergio-teruel** [comment]
@pedrobaeza Ok. What groups can I use here??

**Comment by pedrobaeza** [comment]
For me it should be `base.group_system`

File: agreement/models/agreement.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Agreement(models.Model):
     _description = 'Agreement'
     _inherit = ["mail.thread", "mail.activity.mixin"]
 
-    code = fields.Char(required=True, copy=False)
+    code = fields.Char(required=True)
```
Comment (alexis-via) [inline agreement/models/agreement.py:13]: why did you do that ? There is a unicity constrain...

Comment (sergio-teruel) [reply agreement/models/agreement.py:13]: If field is required you can not duplicate record. Maybe it would be necessary to create a sequence..

**Comment by carlosdauden** [comment]
Another option could be to add "(Copy)" to the code:
https://github.com/odoo/odoo/blob/c5a8051a2c840b2be545eee2f173261a908de737/addons/account/models/account.py#L253
:sweat_smile:

File: agreement/models/agreement.py (hunk 2 above)
Comment (carlosdauden) [reply agreement/models/agreement.py:13]: Another option could be to add "(Copy)" to the code:
https://github.com/odoo/odoo/blob/c5a8051a2c840b2be545eee2f173261a908de737/addons/account/models/account.py#L253
:sweat_smile:

**Comment by sergio-teruel** [comment]
Changes done!!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**